File: robot_examples/haskell-servant/rest_api/lib/Model/RobotInfo.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

module Model.RobotInfo
    ( RobotInfo (..)
    ) where

import Data.Aeson
import GHC.Generics
import Test.QuickCheck
import Model.Number


data RobotInfo = RobotInfo
    { robotId :: Number
    , posX :: Float
    , posY :: Float
    , direction :: Float
    , currentSpeed :: Float
    , requiredSpeed :: Float
    , acceleration :: Float
    , reloadingTime :: Float
    , health :: Float
    } deriving (Show, Eq, Generic)

instance FromJSON RobotInfo
instance ToJSON RobotInfo
instance Arbitrary RobotInfo where
    arbitrary = RobotInfo <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary

File: robot_examples/go/rest_api/swagger/RobotInfo.go
package swagger

import (
)

type RobotInfo struct {
    RobotId  Number  `json:"robotId,omitempty"`
    PosX  float32  `json:"posX,omitempty"`
    PosY  float32  `json:"posY,omitempty"`
    Direction  float32  `json:"direction,omitempty"`
    CurrentSpeed  float32  `json:"currentSpeed,omitempty"`
    RequiredSpeed  float32  `json:"requiredSpeed,omitempty"`
    Acceleration  float32  `json:"acceleration,omitempty"`
    ReloadingTime  float32  `json:"reloadingTime,omitempty"`
    Health  float32  `json:"health,omitempty"`
    
}
